purchases/src/main/kotlin/com/revenuecat/purchases/attributes/SubscriberAttribute.kt

```diff
@@ -0,0 +1,51 @@
+package com.revenuecat.purchases.attributes
+
+import com.revenuecat.purchases.getNullableString
+import com.revenuecat.purchases.utils.DateProvider
+import com.revenuecat.purchases.utils.DefaultDateProvider
+import org.json.JSONException
+import org.json.JSONObject
+import java.util.Date
+
+internal const val JSON_NAME_KEY = "key"
+internal const val JSON_NAME_VALUE = "value"
+internal const val JSON_NAME_SET_TIME = "set_time"
+internal const val JSON_NAME_IS_SYNCED = "is_synced"
+
+internal const val BACKEND_NAME_VALUE = "value"
+internal const val BACKEND_NAME_TIMESTAMP = "updated_at"
+
+internal data class SubscriberAttribute(
+    val key: SubscriberAttributeKey,
+    val value: String?,
+    val dateProvider: DateProvider = DefaultDateProvider(),
+    val setTime: Date = dateProvider.now,
+    val isSynced: Boolean = false
+) {
+
+    constructor(
+        key: String,
+        value: String?,
+        dateProvider: DateProvider = DefaultDateProvider(),
+        setTime: Date = dateProvider.now,
+        isSynced: Boolean = false
+    ) : this(key.getSubscriberAttributeKey(), value, setTime = setTime, isSynced = isSynced)
+
+    @Throws(JSONException::class)
+    constructor(jsonObject: JSONObject) : this(
+        key = jsonObject.getString(JSON_NAME_KEY).getSubscriberAttributeKey(),
+        value = jsonObject.getNullableString(JSON_NAME_VALUE),
+        setTime = Date(jsonObject.getLong(JSON_NAME_SET_TIME)),
+        isSynced = jsonObject.getBoolean(JSON_NAME_IS_SYNCED)
+    )
+
+    fun toJSONObject() = JSONObject().apply {
+        put(JSON_NAME_KEY, key.serverValue)
+        value?.let { put(JSON_NAME_VALUE, value) } ?: put(JSON_NAME_VALUE, JSONObject.NULL)
+        put(JSON_NAME_SET_TIME, setTime.time)
```

Comment:
Is this in milliseconds? The server expects an epoch in seconds (sorry, kind of a pain for Java/Kotlin, but I kept it in seconds since all the Python APIs are seconds based and iOS APIs also use seconds [with decimals]).

Reply:
Good catch. It is fine we use milliseconds in this spot of the code because it's what I use for the cache, but in `toJSONMap()` I was actually sending a String. I created an extension function on Date so that it can be used in other spots. Should I be sending the seconds with decimals too? or long is fine?

Reply:
Long is totally fine!